In [1]:
import sys
import glob
 
sys.path.insert(0, '/Users/rekerl/Documents/Programs/gpc/')
 
from genophenocorr import *


# Full Test of Genotype/Phenotype Correlation Tool

### Upload test patients

In [2]:
allPatients = Cohort('../phenopackets/cohort/*.json', ref = 'hg37')

INFO:pyensembl.sequence_data:Loaded sequence dictionary from /Users/rekerl/Library/Caches/pyensembl/GRCh37/ensembl75/Homo_sapiens.GRCh37.75.cdna.all.fa.gz.pickle
INFO:pyensembl.sequence_data:Loaded sequence dictionary from /Users/rekerl/Library/Caches/pyensembl/GRCh37/ensembl75/Homo_sapiens.GRCh37.75.ncrna.fa.gz.pickle
INFO:pyensembl.sequence_data:Loaded sequence dictionary from /Users/rekerl/Library/Caches/pyensembl/GRCh37/ensembl75/Homo_sapiens.GRCh37.75.pep.all.fa.gz.pickle


No Protein found with ID ENSP00000340373
WARNING - Current Protein ID or Transcript ID does not match with Described Isoform
WARNING - Current Protein ID or Transcript ID does not match with Described Isoform
WARNING - Current Protein ID or Transcript ID does not match with Described Isoform
No Protein found with ID ENSP00000319254
WARNING - Current Protein ID or Transcript ID does not match with Described Isoform
WARNING - Current Protein ID or Transcript ID does not match with Described Isoform
WARNING - Current Protein ID or Transcript ID does not match with Described Isoform
No Protein found with ID ENSP00000371975
No Protein found with ID ENSP00000261517
No Protein found with ID ENSP00000381234
No Protein found with ID ENSP00000371231
No Protein found with ID ENSP00000261517
WARNING - Current Protein ID or Transcript ID does not match with Described Isoform
WARNING - Current Protein ID or Transcript ID does not match with Described Isoform
WARNING - Current Protein ID or Transcrip

### Pull out all diseases with too few patients for testing purposes

In [3]:
byDisease = allPatients.split_by_disease()

In [4]:
too_small = []
for d, dis in byDisease.items():
    if dis.count_patients < 10:
        too_small.append(d)
        
for i in too_small:
    byDisease.pop(i)
print(byDisease.keys())

dict_keys(['OMIM:617425', 'OMIM:616900', 'OMIM:616907', 'OMIM:616975', 'OMIM:159550'])


### Run basic Missense mutation vs Nonsense mutation

In [7]:
for d, dis in byDisease.items():
    result = run_stats(dis, is_var_type, is_var_type, 'missense', 'nonsense')
    print(result.head(5))


Total hpo terms: 171
                                     1 w/ hpo  1 w/o hpo  2 w/ hpo  2 w/o hpo  \
HP:0001880-Eosinophilia                   2.0        8.0       0.0        0.0   
HP:0000347-Micrognathia                   1.0        9.0       0.0        0.0   
HP:0003690-Limb muscle weakness           1.0        9.0       0.0        0.0   
HP:0002867-Abnormality of the ilium       1.0        9.0       0.0        0.0   
HP:0001265-Hyporeflexia                   1.0        9.0       0.0        0.0   

                                     pval  bonferr pval  
HP:0001880-Eosinophilia               1.0           1.0  
HP:0000347-Micrognathia               1.0           1.0  
HP:0003690-Limb muscle weakness       1.0           1.0  
HP:0002867-Abnormality of the ilium   1.0           1.0  
HP:0001265-Hyporeflexia               1.0           1.0  
Total hpo terms: 130
                                                    1 w/ hpo  1 w/o hpo  \
HP:0001252-Muscular hypotonia                   

### Pull out all patients with disease OMIM:616907, print Patient ID and the top effect on the protein

In [8]:
patswith = byDisease['OMIM:616975']
for p, pat in patswith.all_patients.items():
    print(p)
    #print(pat.variant.top_effect.short_description)
    #print(pat.protein)

PMID:27087320-Fregeau-2016-RERE-Subject_1
PMID:27087320-Fregeau-2016-RERE-Subject_7
PMID:29330883-Jordan-2018-RERE-Subject_6
PMID:27087320-Fregeau-2016-RERE-Subject_10
PMID:29330883-Jordan-2018-RERE-Subject_7
PMID:27087320-Fregeau-2016-RERE-Subject_6
PMID:29330883-Jordan-2018-RERE-Subject_4
PMID:27087320-Fregeau-2016-RERE-Subject_5
PMID:29330883-Jordan-2018-RERE-Subject_8
PMID:27087320-Fregeau-2016-RERE-Subject_9
PMID:27087320-Fregeau-2016-RERE-Subject_8
PMID:29330883-Jordan-2018-RERE-Subject_9
PMID:27087320-Fregeau-2016-RERE-Subject_4
PMID:29330883-Jordan-2018-RERE-Subject_5
PMID:29330883-Jordan-2018-RERE-Subject_2
PMID:27087320-Fregeau-2016-RERE-Subject_3
PMID:27087320-Fregeau-2016-RERE-Subject_2
PMID:29330883-Jordan-2018-RERE-Subject_3


### Grabbed an example patient to look at protein features

In [9]:
test = patswith.all_patients.get('PMID:29330883-Jordan-2018-RERE-Subject_6')

patswith.all_proteins

#patswith.all_proteins['ENSP00000338629']
test.variant.variant

Variant(contig='1', start=8418292, ref='G', alt='A', reference_name='GRCh37')

In [10]:
byProteins = allPatients.split_by_protein()

### Run a test that compares HPO terms vs the variant being in one or the other protein domain

In [11]:
result = compare.RunStats(patswith, CF.in_domain, CF.in_domain, [1, 273], [466, 651], 50)

#Print out all significant or lowest 5 

AttributeError: module 'genophenocorr.compare' has no attribute 'RunStats'

In [12]:
for person in byProteins.values():
    print(person)

In [13]:
allPatients.count_vars_per_feature()
#patswith.all_v
#import pandas as pd
#prot = patswith.all_proteins.get('ENSP00000338629')


type  \
Protein         Feature ID                                                   
ENSP00000375704 Domain: Pyrin                                       Domain   
                Domain: NACHT                                       Domain   
ENSP00000264657 Domain: SH2                                         Domain   
                Motif: Essential for nuclear import                  Motif   
ENSP00000367872 Repeat: WD 1                                        Repeat   
...                                                                    ...   
ENSP00000264079 Region: Extracellular/lumenal pore loop             Region   
                Region: Required for palmitoylation and associa...  Region   
                Motif: Dileucine motif; mediates targeting to l...   Motif   
                Motif: Selectivity filter                            Motif   
                Motif: Dileucine internalization motif; mediate...   Motif   

                                                                   start  end  \
Protein         Feature ID                                                      
ENSP00000375704 Domain: Pyrin                                          1   91   
                Domain: NACHT                                        218  534   
ENSP00000264657 Domain: SH2                                          580  670   
                Motif: Essential for nuclear import                  150  162   
ENSP00000367872 Repeat: WD 1                                          53   83   
...                                                                  ...  ...   
ENSP00000264079 Region: Extracellular/lumenal pore loop              107  121   
                Region: Required for palmitoylation and associa...   565  567   
                Motif: Dileucine motif; mediates targeting to l...    11   16   
                Motif: Selectivity filter                            469  474   
                Motif: Dileucine internalization motif; mediate...   573  578   

                                                                    variants  
Protein         Feature ID                                                    
ENSP00000375704 Domain: Pyrin                                             32  
                Domain: NACHT                                             72  
ENSP00000264657 Domain: SH2                                               11  
                Motif: Essential for nuclear import                        8  
ENSP00000367872 Repeat: WD 1                                              16  
...                                                                      ...  
ENSP00000264079 Region: Extracellular/lumenal pore loop                    2  
                Region: Required for palmitoylation and associa...         0  
                Motif: Dileucine motif; mediates targeting to l...         3  
                Motif: Selectivity filter                                  1  
                Motif: Dileucine internalization motif; mediate...         0  

[1865 rows x 4 columns]

In [14]:
#patswith.all_proteins.get('ENSP00000338629').features
for i in allPatients.all_proteins.values():
    try:
        print(i._protein_json.get('comments'))
    except:
        print(i.label)

[]
[{'commentType': 'ALTERNATIVE PRODUCTS', 'events': ['Alternative splicing'], 'isoforms': [{'name': {'value': 'Long'}, 'isoformIds': ['Q92947-1'], 'isoformSequenceStatus': 'Displayed'}, {'name': {'value': 'Short'}, 'isoformIds': ['Q92947-2'], 'sequenceIds': ['VSP_000145'], 'isoformSequenceStatus': 'Described'}]}]
[{'commentType': 'ALTERNATIVE PRODUCTS', 'events': ['Alternative splicing'], 'isoforms': [{'name': {'value': '1'}, 'isoformIds': ['P40763-1'], 'isoformSequenceStatus': 'Displayed'}, {'name': {'value': 'Del-701'}, 'isoformIds': ['P40763-2'], 'sequenceIds': ['VSP_010474'], 'isoformSequenceStatus': 'Described'}, {'name': {'value': '3'}, 'isoformIds': ['P40763-3'], 'sequenceIds': ['VSP_055918', 'VSP_055919'], 'isoformSequenceStatus': 'Described'}]}]
[]
None
[{'commentType': 'ALTERNATIVE PRODUCTS', 'events': ['Alternative splicing'], 'isoforms': [{'name': {'value': '1'}, 'isoformIds': ['P62873-1'], 'isoformSequenceStatus': 'Displayed'}, {'name': {'value': '2'}, 'isoformIds': ['P6

In [15]:
missing = allPatients.all_proteins.get('ENSP00000262097')

In [16]:
missing._protein_json

In [17]:
for i in allPatients.all_proteins.values():
    print(i.id)
    print(i.features)

#allPatients.all_proteins.get('ENSP00000325527').features

ENSP00000375704
                 type start  end
Domain: Pyrin  Domain     1   91
Domain: NACHT  Domain   218  534
ENSP00000222214
Empty DataFrame
Columns: []
Index: []
ENSP00000264657
                                       type start  end
Domain: SH2                          Domain   580  670
Motif: Essential for nuclear import   Motif   150  162
ENSP00000361562
Empty DataFrame
Columns: []
Index: []
ENSP00000340373
Empty DataFrame
Columns: []
Index: []
ENSP00000367872
                type start  end
Repeat: WD 1  Repeat    53   83
Repeat: WD 2  Repeat    95  125
Repeat: WD 3  Repeat   141  170
Repeat: WD 4  Repeat   182  212
Repeat: WD 5  Repeat   224  254
Repeat: WD 6  Repeat   268  298
Repeat: WD 7  Repeat   310  340
ENSP00000256646
                                            type start   end
Domain: EGF-like 1                        Domain    26    63
Domain: EGF-like 2                        Domain    64   102
Domain: EGF-like 3                        Domain   105   143
Domain: EG